open-metadata-implementation/access-services/asset-lineage/asset-lineage-server/src/main/java/org/odpi/openmetadata/accessservices/assetlineage/listeners/AssetLineageOMRSTopicListener.java

```diff
@@ -167,39 +165,62 @@ private void processDeletedEntity(EntityDetail entityDetail) throws ConnectorChe
     private void processClassifiedEntityEvent(EntityDetail entityDetail) throws OCFCheckedExceptionBase, JsonProcessingException {
         if (!immutableValidLineageEntityEvents.contains(entityDetail.getType().getTypeDefName()))
             return;
-        log.debug("Asset Lineage OMAS is processing a Classified Entity event which contains the following entity {}: ", entityDetail.getGUID());
+        log.debug(PROCESSING_ENTITYDETAIL_DEBUG_MESSAGE, "classifiedEntity", entityDetail.getGUID());
         publisher.publishClassificationContext(entityDetail);
     }
 
-    private void processReclassifiedEntityEvent(EntityDetail entityDetail) {
-        log.debug("Asset Lineage OMAS is processing a ReClassified Entity event which contains the following entity {}: ", entityDetail.getGUID());
+    private void processReclassifiedEntityEvent(EntityDetail entityDetail) throws OCFCheckedExceptionBase, JsonProcessingException {
+        if (!immutableValidLineageEntityEvents.contains(entityDetail.getType().getTypeDefName()))
+            return;
+        log.debug(PROCESSING_ENTITYDETAIL_DEBUG_MESSAGE, "reclassifiedEntity", entityDetail.getGUID());
+        publisher.publishClassificationContext(entityDetail);
     }
 
-    private void processDeclassifiedEntityEvent(EntityDetail entityDetail) {
-        log.debug("Asset Lineage OMAS is processing a DeClassified Entity event which contains the following entity {}: ", entityDetail.getGUID());
+    private void processDeclassifiedEntityEvent(EntityDetail entityDetail) throws OCFCheckedExceptionBase, JsonProcessingException {
+        if (!immutableValidLineageEntityEvents.contains(entityDetail.getType().getTypeDefName()))
+            return;
+        log.debug(PROCESSING_ENTITYDETAIL_DEBUG_MESSAGE, "declassifiedEntity", entityDetail.getGUID());
+        if (anyValidClassificationsLeft(entityDetail)) {
+            publisher.publishClassificationContext(entityDetail);
+            return;
+        }
+        //The last relevant lineage classification has been removed from the entity
+        LineageEvent event = new LineageEvent();
+        event.setLineageEntity(converter.createLineageEntity(entityDetail));
+        event.setAssetLineageEventType(AssetLineageEventType.DECLASSIFIED_ENTITY_EVENT);
+        publisher.publishEvent(event);
     }
 
+
     private void processNewRelationship(Relationship relationship) {
-        log.debug(PROCESSING_RELATIONSHIP_DEBUG_MESSAGE, "NewRelationship", relationship.getGUID());
+        log.debug(PROCESSING_RELATIONSHIP_DEBUG_MESSAGE, "newRelationship", relationship.getGUID());
     }
 
     private void processUpdatedRelationshipEvent(Relationship relationship) throws OCFCheckedExceptionBase, JsonProcessingException {
-        log.debug(PROCESSING_RELATIONSHIP_DEBUG_MESSAGE, AssetLineageEventType.UPDATE_RELATIONSHIP_EVENT, relationship.getGUID());
+        log.debug(PROCESSING_RELATIONSHIP_DEBUG_MESSAGE, "updatedRelationship", relationship.getGUID());
```

Comment:
Just a question here, why don't you want to use directly the event type(e..g AssetLineageEventType.UPDATE_RELATIONSHIP_EVENT) for debugging? And use another string instead?

Reply:
AssetLineageEventType does not refer to the input events of the omas, but to its output events. Some input events, such as classify and reclassify, may lead to an event of the same AssetLineageEventType (CLASSIFICATION_CONTEXT). Therefore these constants are not sutied for classifying the input events